### ISSUE TITLE
amplitude-js there is no way to pass the insert_id parameter

### ISSUE THREAD

**Issue by dulikmixail** [amplitude/Amplitude-Node#82]
I use "amplitude-js": "^7.4.3". Using this version, I found that it is not possible to pass the "insert_id" parameter, but it is described in the HTTP API V2 and you recommend passing it!! How to pass this parameter?

**Comment by kelvin-lu** [comment]
To be clear this is an issue filed for the [Browser-based Javascript SDK](https://github.com/amplitude/Amplitude-JavaScript) and not the server-side Node SDK?

We didn't have optioning on that SDK for insert ID - instead, we will generate a UUID for each `logEvent` so that it's always attached and has a unique identifier in accordances with the HTTP v2.

**Comment by dulikmixail** [comment]
Yes, this is an issue filled for the [Browser-based Javascript SDK](https://github.com/amplitude/Amplitude-JavaScript). Because i want to use "insert_id" parameter so that when the event is re-sent so that it is displayed only once in amplitude per day

**Comment by kelvin-lu** [comment]
@dulikmixail sorry for the late response here!

When you say "event is resent", do you mean you are considering when events fail (ex. network offline) and need to be tried again or do you mean that your implementation may call `logEvent` twice with logically the same event and you want to pass your own insert_id ?

**Comment by lynnshaoyu** [comment]
hey kelvin, so are you saying there is no insert_ID property for the javascript SDK? i'm wondering if this is the case as well for the typescript SDK.

**Comment by stipelelas** [comment]
Is there really no way to use `insert_id` with the Browser or Node SDK? Does it only work with the HTTP V2 API?

I'm trying to send an event like this:

```
const eventProperties = {
  randomProperty: '1234',
  insert_id: 'abcd1234'
};

amplitude.track('Test event', eventProperties);
```

Will this event be successfully deduplicated?
